src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/MigrationIncentivesMessageListener.kt

```diff
@@ -18,37 +22,46 @@ import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.incentives.Incent
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.listeners.context
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.INCENTIVES_QUEUE_ID
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationMessage
+import java.util.concurrent.CompletableFuture
 
 @Service
 class MigrationIncentivesMessageListener(
   private val objectMapper: ObjectMapper,
   private val incentivesMigrationService: IncentivesMigrationService,
-  private val incentivesSynchronisationService: IncentivesSynchronisationService
+  private val incentivesSynchronisationService: IncentivesSynchronisationService,
 ) {
 
   private companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  @JmsListener(destination = INCENTIVES_QUEUE_ID, containerFactory = "hmppsQueueContainerFactoryProxy", concurrency = "5")
-  fun onMessage(message: String, rawMessage: SQSTextMessage) {
+  @OptIn(DelicateCoroutinesApi::class)
+  @SqsListener(INCENTIVES_QUEUE_ID, factory = "hmppsQueueContainerFactoryProxy")
+  fun onMessage(message: String, rawMessage: Message): CompletableFuture<Void> {
     log.debug("Received message {}", message)
     val migrationMessage: MigrationMessage<IncentiveMessages, *> = message.fromJson()
-    kotlin.runCatching {
-      when (migrationMessage.type) {
-        MIGRATE_INCENTIVES -> incentivesMigrationService.divideIncentivesByPage(context(message.fromJson()))
-        MIGRATE_INCENTIVES_BY_PAGE -> incentivesMigrationService.migrateIncentivesForPage(context(message.fromJson()))
-        MIGRATE_INCENTIVE -> incentivesMigrationService.migrateIncentive(context(message.fromJson()))
-        MIGRATE_INCENTIVES_STATUS_CHECK -> incentivesMigrationService.migrateIncentivesStatusCheck(context(message.fromJson()))
-        CANCEL_MIGRATE_INCENTIVES -> incentivesMigrationService.cancelMigrateIncentivesStatusCheck(context(message.fromJson()))
-        RETRY_INCENTIVE_MAPPING -> incentivesMigrationService.retryCreateIncentiveMapping(context(message.fromJson()))
-        SYNCHRONISE_CURRENT_INCENTIVE -> incentivesSynchronisationService.handleSynchroniseCurrentIncentiveMessage(context(message.fromJson()))
-        RETRY_INCENTIVE_SYNCHRONISATION_MAPPING -> incentivesSynchronisationService.retryCreateIncentiveMapping(context(message.fromJson()))
+    return GlobalScope.launch {
+      runCatching {
```

Comment:
Is GlobalScope.launch ok here? We're only kicking off one thread which seems to mitigate the warnings about structured concurrency and leakages...